### PR TITLE
MLS Importer

### PR DESCRIPTION
**Pull request by eziolotta**
Importer for Multilingual LibriSpeech (MLS) Dataset  - ticket [114](https://github.com/MozillaItalia/DeepSpeech-Italian-Model/issues/114)

In corpora_importer.py,  are exported in csv output information on duration and comments of audio file (author, title, etc.) 
A filter method, based on comments,  has been implemented , but this system has only been partially tested.

Filtering by author in MLS dataset:
Given the long execution times of importer, and possibility of adding authors only later, this operation could also be done in the corpora_collector

### CODE REVIEW

File: MITADS-Speech/corpora_importer.py [hunk 1]
```diff
@@ -177,22 +181,32 @@ def _maybe_convert_wav(self,mp3_filename, wav_filename):
...
+        mp3_wav_filename = sample[0]
+        make_wav_resample = sample[1]
+        # Storing wav files next to the audio filename ones - just with a different suffix
         wav_filename = path.splitext(mp3_wav_filename)[0] + ".wav"
-        self._maybe_convert_wav(mp3_wav_filename, wav_filename)            
+
+        ##Note: to get frames/duration for mp3/wav audio we not use soxi command but sox.file_info.duration(
+        ##soxi command is not present in Windows sox distribution  - see this  https://github.com/rabitt/pysox/pull/74
+        duration = sox.file_info.duration(mp3_wav_filename)
+        comments = sox.file_info.comments(mp3_wav_filename)
```
Comment (nefastosaturo) [inline MITADS-Speech/corpora_importer.py:199]: pySox does not handle char encoding, so it raise an error when trying to decode bytes to utf8
```suggestion
        comments = ""
        try:
          comments=sox.file_info.comments(mp3_wav_filename)
        except UnicodeError as e:
          try:
            completedProcess=subprocess.run(["soxi", "-a", mp3_wav_filename], stdout=subprocess.PIPE)
            comments=completedProcess.stdout.decode("utf-8", "ignore")
          except:
            pass
```

File: MITADS-Speech/corpora_importer.py [hunk 2]
```diff
@@ -122,9 +123,18 @@ def _maybe_extract(self,target_dir, extract_dir, archive_path):
...
+            ##check file if zip or tar
+            if(archive_path.endswith('.zip')):
+                ##extraxt zip file
+                with ZipFile(archive_path, "r") as zipobj:
+                    # Extract all the contents of zip file in current directory
+                    zipobj.extractall(target_dir)
+            else:
+                ##extract other gzip, bz2 and lzma
+                tar = tarfile.open(archive_path)
+                tar.extractall(extracted_path)
```
Comment (nefastosaturo) [inline MITADS-Speech/corpora_importer.py:136]: child folder with same name of its parent (eg: origin/mls_italian/mls_italian)
```suggestion
                tar.extractall(target_dir)
```

File: MITADS-Speech/corpora_importer.py [hunk 3]
```diff
@@ -141,32 +151,26 @@ def _maybe_convert_sets(self,corpus:Corpus):
...
+            bar.update(i)
+        bar.update(num_samples)
+        pool.close()
+        pool.join()        
+       
+        ########################################
+        ## filtered rows data are evaluated in write_csv
+        self._write_csv(corpus,rows)
 
     def _maybe_convert_wav(self,mp3_filename, wav_filename):
```
Comment (nefastosaturo) [inline MITADS-Speech/corpora_importer.py:175]: in this function there is a typo: CHANNELS must be N_CHANNELS

**Comment by nefastosaturo** [comment]
oh, btw, I got problem installing the "pycopy-shutil" in colab but maybe its just a google colab problem. I have just bypassed it without any problem

File: MITADS-Speech/corpora_importer.py [hunk 4]
```diff
@@ -231,15 +245,30 @@ def _write_csv(self,corpus:Corpus):
...
+            
+            duration = row_data[3]
+            comments = row_data[4]
+
             st = os.stat(_file)
             file_size = st.st_size
             utterence = utterences[_file]
+
             utterence_clean = utterence      
+            speaker_id = corpus.get_speaker_id(_file)
```
Comment (nefastosaturo) [inline MITADS-Speech/corpora_importer.py:266]: Where `get_speaker_id` is defined?

File: MITADS-Speech/corpora_importer.py (hunk 3 above)
Comment (eziolotta) [reply MITADS-Speech/corpora_importer.py:175]: yes, i fix!

File: MITADS-Speech/corpora_importer.py (hunk 4 above)
Comment (eziolotta) [reply MITADS-Speech/corpora_importer.py:274]: sorry I forgot to add the implemented method i Corpus class, without this going into error. However it is not used in MLS importer yet.

**Comment by eziolotta** [comment]
I will have to run MLS importer and also the other importers (maybe partially...) , for  code changes  on corpora_importer

**Comment by nefastosaturo** [comment]
> I will have to run MLS importer and also the other importer (maybe partially...) , to test code changes on corpora_importer

yep, i'm testing all the importers right now

**Pull request merged**